lambdas/tests/unit/services/test_oidc_service.py

```diff
@@ -0,0 +1,238 @@
+    mocker, oidc_service
+):
+    mock_access_token = "mock_access_token"
+    invalid_id_token = "invalid_id_token"
+    mock_cis2_response = MockResponse(
+        status_code=200,
+        json_data={
+            "access_token": mock_access_token,
+            "scope": "openid",
+            "id_token": invalid_id_token,
+            "token_type": "Bearer",
+            "expires_in": 3599,
+        },
+    )
+    mocker.patch("requests.post", return_value=mock_cis2_response)
+    mocker.patch.object(
+        oidc_service, "validate_and_decode_token", side_effect=AuthorisationException
+    )
+
+    with pytest.raises(AuthorisationException):
+        oidc_service.fetch_tokens("test_auth_code")
+
+
+def test_oidc_service_fetch_user_org_codes(mocker, oidc_service):
+    mock_token = "fake_access_token"
+    mock_userinfo = {
+        "nhsid_useruid": "500000000000",
+        "name": "TestUserOne Caius Mr",
+        "nhsid_nrbac_roles": [
+            {
+                "person_orgid": "500000000000",
+                "person_roleid": "500000000000",
+                "org_code": "A9A5A",
+                "role_name": '"Support":"Systems Support":"Systems Support Access Role"',
+                "role_code": "S8001:G8005:R8015",
+            },
+            {
+                "person_orgid": "500000000000",
+                "person_roleid": "500000000000",
+                "org_code": "B9A5A",
+                "role_name": '"Primary Care Support England":"Systems Support Access Role"',
+                "role_code": "S8001:G8005:R8015",
+            },
+        ],
+        "given_name": "Caius",
+        "family_name": "TestUserOne",
+        "uid": "500000000000",
+        "nhsid_user_orgs": [
+            {"org_name": "NHSID DEV", "org_code": "A9A5A"},
+            {"org_name": "Primary Care Support England", "org_code": "B9A5A"},
+        ],
+        "sub": "500000000000",
+    }
+
+    expected = ["A9A5A", "B9A5A"]
+
+    mock_response = MockResponse(status_code=200, json_data=mock_userinfo)
+
+    mocker.patch("requests.get", return_value=mock_response)
+
+    actual = oidc_service.fetch_user_org_codes(mock_token)
+    assert actual == expected
+
+
+def test_oidc_service_fetch_user_org_codes_raise_AuthorisationException_for_invalid_access_token(
+    mocker, oidc_service
+):
+    mock_token = "fake_access_token"
+
+    mock_response = MockResponse(
+        status_code=401,
+        json_data={
+            "error_description": "The access token provided is expired, revoked, "
+            "malformed, or invalid for other reasons.",
+            "error": "invalid_token",
+        },
+    )
+    mocker.patch("requests.get", return_value=mock_response)
+
+    with pytest.raises(AuthorisationException):
+        oidc_service.fetch_user_org_codes(mock_token)
+
+
+# Below are tests around the validate_and_decode_token method. Involve mocking the CIS2 signing key pairs
```

Comment:
I'd personally avoid comments like this, code should generally speak for itself rather than littering the code base with little comments like this unless it's obscure?

Reply:
Thanks! Removed the redundant comment.